### PR TITLE
Bitbar/tunnel

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

Adds tunnel starting/stopping to BitBar utils and hooks, modifying previously established BitBar options to accommodate it.

## TODO:
- Add unit tests to tunnel start util

### CODE REVIEW

File: dockerfiles/Dockerfile.ci [hunk 1]
```diff
@@ -16,6 +16,9 @@ RUN wget -q https://saucelabs.com/downloads/sc-4.6.4-linux.tar.gz \
   && rm sc-4.6.4-linux.tar.gz \
   && mv sc-4.6.4-linux sauce-connect
 
+RUN wget -q https://sbsecuretunnel.s3.amazonaws.com/cli/macos/SBSecureTunnel \
```
Comment (twometresteve) [inline dockerfiles/Dockerfile.ci:19]: I'm guessing this should be the Linux CLI for the Docker file, not macos?

File: lib/maze/bitbar_utils.rb [hunk 2]
```diff
@@ -47,6 +50,35 @@ def upload_app(api_key, app)
...
+      # @param username [String] Username to start the tunnel with
+      # @param access_key [String] BitBar access key
+      def start_local_tunnel(bb_local, username, access_key)
+        $logger.info 'Starting BitBar SBSecureTunnel local tunnel'
+        command = "#{bb_local} --username #{username} --authkey #{access_key}" \
+                    "--ready #{BB_READY_FILE} --kill #{BB_KILL_FILE}"
+
+        @tunnel_shell = Maze::InteractiveCLI.new
+        @tunnel_shell.run_command(command)
+        success = Maze::Wait.new(timeout: 30).until do
```
Comment (twometresteve) [inline lib/maze/bitbar_utils.rb:65]: Is 30 seconds likely to be plenty?  Of course there's a balance, though - we don't want to wait too long if it's a genuine failure case.

File: lib/maze/bitbar_utils.rb [hunk 3]
```diff
@@ -47,6 +50,35 @@ def upload_app(api_key, app)
...
+          File.exist?(BB_READY_FILE)
+        end
+        unless success
+          $logger.info "Failed: #{@tunnel_shell.stdout_lines}"
```
Comment (twometresteve) [inline lib/maze/bitbar_utils.rb:69]:
```suggestion
          $logger.error "Failed: #{@tunnel_shell.stdout_lines}"
```

File: lib/maze/bitbar_utils.rb [hunk 4]
```diff
@@ -47,6 +50,35 @@ def upload_app(api_key, app)
...
+        end
+      end
+
+      # Stops the local tunnel
+      def stop_local_tunnel
+        FileUtils.touch(BB_KILL_FILE)
+        Maze::Wait.new(timeout: 30).until do
+          !@tunnel_shell.running?
+        end
+        File.delete(BB_READY_FILE) if File.exist?(BB_READY_FILE)
```
Comment (twometresteve) [inline lib/maze/bitbar_utils.rb:79]: Is it worth deleting these if they exist before starting the tunnel as well?  I know I can be very Command-C happy, potentially leaving files lying around.

File: lib/maze/option/parser.rb [hunk 5]
```diff
@@ -89,8 +89,8 @@ def parse(args)
                 type: :string
 
             # BitBar-only options
-            opt Option::BITBAR_API_KEY,
-                '(BB only) BitBar API key',
+            opt Option::BB_LOCAL,
+                '(BB only) Path to the SBSecureTunnel binary. MAZE_BB_LOCAL env var or "/sauce-connect/bin/sc" by default',
```
Comment (twometresteve) [inline lib/maze/option/parser.rb:93]:
```suggestion
                '(BB only) Path to the SBSecureTunnel binary. MAZE_BB_LOCAL env var or "/SBSecureTunnel" by default',
```

File: lib/maze/bitbar_utils.rb (hunk 2 above)
Comment (Cawllec) [reply lib/maze/bitbar_utils.rb:65]: From what I can tell it starts pretty quickly, if it goes for 30 seconds without starting it's failed

**Review by twometresteve — APPROVED** [review]
LGTM

**Pull request merged**